at_secondary/at_secondary_server/lib/src/connection/outbound/outbound_message_listener.dart

```diff
@@ -82,17 +82,29 @@ class OutboundMessageListener {
           // Right now, all callers of this method only expect there ever to be a 'data:' response.
           // So right now, the right thing to do here is to throw an exception.
           // We can leave the connection open since an 'error:' response indicates normal functioning on the other end
-          throw AtConnectException("Request to remote secondary ${outboundClient.toAtSign} at ${outboundClient.toHost}:${outboundClient.toPort} received error response '$result'");
+          try {
+            result = result.toString().replaceFirst('error:', '');
+            var errorMap = jsonDecode(result);
+            throw AtExceptionUtils.get(
+                errorMap['errorCode'], errorMap['errorDescription']);
+          } on FormatException {
+            // Catching the FormatException to preserve backward compatibility - responses without jsonEncoding.
+            // TODO: Can remove the catch block in the next release (once all the existing servers are migrated to new version).
+            throw AtConnectException(
```

Comment:
GIven that the old behaviour was to throw AtConnectException, you need to check for code in the server which was catching AtConnectException, and ensure it works properly with the new exception i.e. whatever `AtExceptionUtils.get` creates